src/modules/analysis/fpsigmatcher/Utils.c

```diff
@@ -144,7 +144,7 @@ char ** get_filenames(const char * directory, int * num_of_files) {
 			/* Skip non-regular files (e.g. directories) */
 			if ( ((fileNameSize < (int)sizeof(filename)) && (fileNameSize > 0))
 					&& (stat(filename, &dirEntStat) == 0) && (S_ISREG(dirEntStat.st_mode)) ) {
-				snprintf(filenames[i], MAX_SIZE_OF_FILENAME-1, "%s", pDirEnt->d_name);
+				snprintf(filenames[i], MAX_SIZE_OF_FILENAME, "%s", pDirEnt->d_name);
```

Comment:
Can you briefly explain this change?

Reply:
gcc complains that one byte is not used here, and it considers that to be a mistake.
snprintf writes at most n bytes, including the '\0', so MAX_SIZE_OF_FILENAME-1 in total, in a buffer "filenames[i]" which has MAX_SIZE_OF_FILENAME bytes.

Reply:
👍